From fc3d4d7d7836148069f54267526abcb039280fec Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 7 Aug 2020 12:52:06 +0000
Subject: [PATCH] sandbox-sched_setscheduler.patch

allow sched_setscheduler in the GMP/RDD sandbox

upstream bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1657849
Co-Authored-By: ptrcnull <git@ptrcnull.me>
---
 security/sandbox/linux/SandboxFilter.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/security/sandbox/linux/SandboxFilter.cpp b/security/sandbox/linux/SandboxFilter.cpp
index 337bce4413f9d..da25127bb85c3 100644
--- a/security/sandbox/linux/SandboxFilter.cpp
+++ b/security/sandbox/linux/SandboxFilter.cpp
@@ -1919,13 +1919,13 @@ class GMPSandboxPolicy : public SandboxPolicyCommon {
         return Allow();
       case __NR_sched_get_priority_min:
       case __NR_sched_get_priority_max:
+      case __NR_sched_setscheduler:
         return Allow();
       // The OpenH264 plugin needs sched_getaffinity() in multithreaded mode;
       // bug 2071378.
       case __NR_sched_getaffinity:
       case __NR_sched_getparam:
-      case __NR_sched_getscheduler:
-      case __NR_sched_setscheduler: {
+      case __NR_sched_getscheduler: {
         Arg<pid_t> pid(0);
         return If(pid == 0, Allow()).Else(Trap(SchedTrap, nullptr));
       }
@@ -2209,12 +2209,14 @@ class RDDSandboxPolicy final : public SandboxPolicyCommon {
       case __NR_sched_getparam:
       case __NR_sched_setparam:
       case __NR_sched_getscheduler:
-      case __NR_sched_setscheduler:
       case __NR_sched_getattr:
       case __NR_sched_setattr: {
         Arg<pid_t> pid(0);
         return If(pid == 0, Allow()).Else(Trap(SchedTrap, nullptr));
       }
+      // sched_setscheduler gets special treatment here (bug 1657849):
+      case __NR_sched_setscheduler:
+        return Allow();
 
         // The priority bounds are also used, sometimes (bug 1838675):
       case __NR_sched_get_priority_min:
-- 
2.56.0

